### PR TITLE
[codegen/go] Fix Input/Output methods for Go resources

### PR DESCRIPTION
**Pull request by lblackstone**

### CODE REVIEW

File: pkg/codegen/go/gen.go [hunk 1]
```diff
@@ -645,7 +649,7 @@ func (pkg *pkgContext) genInputTypes(w io.Writer, t *schema.ObjectType, details
 	}
 	fmt.Fprintf(w, "}\n\n")
 
-	genInputMethods(w, name, name+"Args", name, details.ptrElement)
```
Comment (pgavlin) [inline pkg/codegen/go/gen.go:648]: Should the `resourceType` parameter be `true` here? I think that `genInputTypes` is called for non-resource types as well.

Comment (lblackstone) [reply pkg/codegen/go/gen.go:648]: Oops, I think I meant to default that arg to false. I'll double-check that.

**Comment by lblackstone** [comment]
This may conflict with https://github.com/pulumi/pulumi/pull/5920

**Comment by viveklak** [comment]
@lblackstone is there a way we can test this short of relying on sdk integration tests? Wondering if a test like the one here: https://github.com/pulumi/pulumi/blob/master/pkg/codegen/go/gen_test.go#L114 is feasible to make sure we catch bugs closer to the source? Its part of my frustration with the brittleness of go codegen - things `look` ok but break in subtle ways.

I have another change where all generated test file contents are asserted,  so the `provider.go` is checked in as part of the test suite, e.g.: https://github.com/pulumi/pulumi/pull/5826/files#diff-b9d79a3c3244ac0021fd8dadcf83810a0d8f5572faeea02502277ebf380060ceR1

**Comment by lblackstone** [comment]
> @lblackstone is there a way we can test this short of relying on sdk integration tests?

I updated the static test data to reflect these changes, but that still doesn't check if the changes actually work. It would be really helpful to also have functional validation, so let's try to get that on the roadmap.

**Review by pgavlin — APPROVED** [review]

**Review by viveklak — APPROVED** [review]

**Pull request merged**